CONTRIBUTING.md

```diff
@@ -118,11 +118,39 @@ For example:
 - modify your Python file `torch/__init__.py`
 - test functionality
 
-You do not need to repeatedly install after modifying Python files.
+You do not need to repeatedly install after modifying Python files. Note that this is only true
+after modifying Python files and you would need to reinstall if you modify other files.
 
 In case you want to reinstall, make sure that you uninstall PyTorch first by running `pip uninstall torch`
 and `python setup.py clean`. Then you can install in `develop` mode again.
 
+### Tips and Debugging
+* A prerequisite to installing PyTorch is CMake. We recommend installing it with [Homebrew](https://brew.sh/)
+with `brew install cmake` if you are developing on MacOS or Linux system.
+* Our `setup.py` requires Python 3. Python 2 is deprecated anyway, so you should definitely get with the times
+and install Python 3.
+* If you run into errors when running `python setup.py develop`, here are some debugging steps:
+  1. Run `printf '#include <stdio.h>\nint main() { printf("Hello World");}'|clang -x c -; ./a.out` to make sure
+  your CMake works and can compile this simple Hello World program without errors.
+  2. If you have made edits to the PyTorch repo, commit any change you'd like to keep and clean the repo with the
+  following commands (note that clean _really_ removes all untracked files and changes.):
+  ```bash
+  git submodule deinit -f .
+  git clean -xdf
+  python setup.py clean
+  git submodule update --init --recursive # very important to sync the submodules
+  python setup.py develop                 # then try running the command again
+  ```
+  3. If you still get failures, nuke your `build` directory. The `setup.py` script compiles binaries into the
+  `build` folder and caches many details along the way, which saves time the next time you build. If you're running
+  into issues, you can always `rm -rf build` from the toplevel `pytorch` directory and start over.
+  4. `python setup.py develop` is equivalent to running `make` from the `build` directory. If you want more detailed
```

Comment:
this is not true. I would suggest change it to:
```suggestion
  4. The main step within `python setup.py develop` is running `make` from the `build` directory. If you want more detailed
```

Although I don't think the command below gives more error message. you can also give environment variables via  `ENV_KEY1=ENV_VAL1[, ENV_KEY2=ENV_VAL2]* python setup.py develop`